jolt-core/src/poly/commitment/dory.rs

```diff
@@ -0,0 +1,568 @@
+            .collect(),
+    }
+}
+
+/// Dory commitment scheme implementation
+#[derive(Clone, Debug)]
+pub struct DoryCommitmentScheme<F: JoltField, ProofTranscript: Transcript> {
+    _phantom_field: PhantomData<F>,
+    _phantom_transcript: PhantomData<ProofTranscript>,
+}
+
+/// Setup structure containing both prover and verifier parameters.
+/// Dory API exposes an SRS to disk option, but is not (yet) utilized here.
+#[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DorySetup {
+    pub prover_setup: ProverSetup<ArkBn254Pairing>,
+    pub verifier_setup: VerifierSetup<ArkBn254Pairing>,
+    pub max_log_n: usize,
+}
+
+/// Commitment structure
+#[derive(Clone, Debug, Default, PartialEq, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryCommitment {
+    pub commitment: Fq12, // this is GT
+}
+
+/// Proof structure storing the serializable DoryProof data and other data
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryProof<F: JoltField> {
+    pub evaluation: Fr,
+    pub opening_point: Vec<Fr>,
+    pub sigma: usize,
+    // Store the serializable DoryProof from the DoryProofBuilder
+    pub dory_proof_data: DoryProofData<ark_bn254::G1Affine, dory::curve::G2AffineWrapper, Fq12>,
+    _phantom: PhantomData<F>,
+}
+
+/// Batched proof structure
+/// @TODO: this is not yet fully implemented.
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryBatchedProof<F: JoltField> {
+    proofs: Vec<DoryProof<F>>,
+}
+
+impl<F, ProofTranscript> CommitmentScheme<ProofTranscript>
+    for DoryCommitmentScheme<F, ProofTranscript>
+where
+    F: JoltField,
+    ProofTranscript: Transcript,
+{
+    type Field = F;
+    type Setup = DorySetup;
+    type Commitment = DoryCommitment;
+    type Proof = DoryProof<F>;
+    type BatchedProof = DoryBatchedProof<F>;
+
+    fn setup(max_log_n: usize) -> Self::Setup {
+        let (prover_setup, verifier_setup) = setup::<ArkBn254Pairing, _>(thread_rng(), max_log_n);
+
+        DorySetup {
+            prover_setup,
+            verifier_setup,
+            max_log_n,
+        }
+    }
+
+    fn commit(poly: &MultilinearPolynomial<Self::Field>, setup: &Self::Setup) -> Self::Commitment {
+        // Extract polynomial coefficients
+        let field_coeffs = extract_field_coefficients(poly);
+
+        // Convert JoltField coefficients to Fr
+        // This assumes F is Fr or can be safely transmuted to Fr
+        let coeffs: Vec<Fr> = field_coeffs
+            .iter()
+            .map(|&coeff| unsafe { std::mem::transmute_copy(&coeff) })
+            .collect();
+
+        // For Dory, sigma is the number of columns in the matrix representation
+        // we use sigma = ceil(num_vars / 2) so we have a nice square matrix.
+        let num_vars = poly.get_num_vars();
+        let sigma = (num_vars + 1) / 2;
+        let commitment =
+            commit::<ArkBn254Pairing, OptimizedMsmG1>(&coeffs, 0, sigma, &setup.prover_setup);
```

Comment:
this is really a comment for the other repo, but let's change the output type of `commit` to `PairingOutput<Bn254>`. The (unwrapped) target field type doesn't implement `Mul<P::ScalarField>`, which we need to homomorphically combine Dory commitments